pkg/controllers/management/auth/globalroles/fleetworkspace_binding_handler.go

```diff
@@ -0,0 +1,203 @@
+	globalRole, err := h.grCache.Get(globalRoleBinding.GlobalRoleName)
+	if err != nil {
+		return fmt.Errorf("unable to get globalRole: %w", err)
+	}
+
+	if err = h.reconcileResourceRulesBindings(globalRoleBinding, globalRole); err != nil {
+		return fmt.Errorf("error reconciling fleet permissions rules: %w", err)
+	}
+
+	if err = h.reconcileWorkspaceVerbsBindings(globalRoleBinding, globalRole); err != nil {
+		return fmt.Errorf("error reconciling fleet workspace verbs: %w", err)
+	}
+
+	return nil
+}
+
+func (h *fleetWorkspaceBindingHandler) reconcileResourceRulesBindings(grb *v3.GlobalRoleBinding, gr *v3.GlobalRole) error {
+	fleetWorkspaces, err := h.fwCache.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("unable to list fleetWorkspaces when reconciling globalRoleBinding %s: %w", grb.Name, err)
+	}
+
+	var returnError error
+	for _, fleetWorkspace := range fleetWorkspaces {
+		if fleetWorkspace.Name == localFleetWorkspace {
+			continue
+		}
+		desiredRB := backingRoleBinding(grb, gr, fleetWorkspace.Name)
+		rb, err := h.rbCache.Get(fleetWorkspace.Name, wrangler.SafeConcatName(grb.Name))
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				returnError = multierror.Append(returnError, err)
+				continue
+			}
+			if gr.InheritedFleetWorkspacePermissions.ResourceRules != nil {
+				_, err = h.rbClient.Create(desiredRB)
+				if err != nil {
+					returnError = multierror.Append(returnError, err)
+				}
+			}
+			continue
+		}
+
+		if gr.InheritedFleetWorkspacePermissions.ResourceRules == nil {
+			err := h.rbClient.Delete(rb.Namespace, rb.Name, &metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				return err
+			}
+			return nil
+		}
+		if !reflect.DeepEqual(rb.Subjects, desiredRB.Subjects) ||
+			!reflect.DeepEqual(rb.RoleRef, desiredRB.RoleRef) {
+			// undo modifications if rb has changed.
+			err := h.rbClient.Delete(rb.Namespace, rb.Name, &metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				returnError = multierror.Append(returnError, err)
+			}
+			_, err = h.rbClient.Create(desiredRB)
+			if err != nil {
+				returnError = multierror.Append(returnError, err)
+			}
+		}
+	}
+
+	return returnError
+}
+
+func (h *fleetWorkspaceBindingHandler) reconcileWorkspaceVerbsBindings(grb *v3.GlobalRoleBinding, gr *apimgmtv3.GlobalRole) error {
+	crbName := wrangler.SafeConcatName(grb.Name, fleetWorkspaceVerbsName)
+	desiredCRB := backingClusterRoleBinding(grb, gr, crbName)
+
+	crb, err := h.crbCache.Get(crbName)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		if gr.InheritedFleetWorkspacePermissions.ResourceRules != nil {
+			_, err = h.crbClient.Create(desiredCRB)
+			return err
+		}
+		return nil
+	}
+
+	if gr.InheritedFleetWorkspacePermissions.ResourceRules == nil {
+		return h.crbClient.Delete(crbName, &metav1.DeleteOptions{})
```

Comment:
Like in other cases, I think that we want to check for a not found error here. Otherwise, this could cause the controller to infinitely loop if the CRB is deleted before this function is called.